pkg/inject/sidecar.go

```diff
@@ -211,6 +211,7 @@ func container(jaeger *v1.Jaeger, dep *appsv1.Deployment) corev1.Container {
 	// see https://github.com/jaegertracing/jaeger-operator/issues/334
 	sort.Strings(args)
 
+	dep.Spec.Template.Spec.ImagePullSecrets = util.RemoveDuplicatedImagePullSecrets(append(dep.Spec.Template.Spec.ImagePullSecrets, commonSpec.ImagePullSecrets...))
```

Comment:
Hm now that you mention that, I wonder if we should change
`dep.Spec.Template.Spec.ImagePullSecrets = util.RemoveDuplicatedImagePullSecrets(append(dep.Spec.Template.Spec.ImagePullSecrets, commonSpec.ImagePullSecrets...))`
to
`dep.Spec.Template.Spec.ImagePullSecrets = util.RemoveDuplicatedImagePullSecrets(append(dep.Spec.Template.Spec.ImagePullSecrets, jaeger.Spec.Agent.JaegerCommonSpec.ImagePullSecrets...))`

i.e., don't use the imagePullSecrets from the commonSpec, but rather only the ones under the agent spec, which would then be more justified since it would seem to be a more intentional decision.

Reply:
It should be a user documentation, probably under the sidecar part: https://www.jaegertracing.io/docs/1.18/operator/#auto-injecting-jaeger-agent-sidecars